src/kernel/thread.c

```diff
@@ -131,14 +132,19 @@ void init_thread(Thread* thread, int tid, const char* name,
   init_register_context(thread);
 }
 
-__attribute__((constructor)) static void init_threads(void) {
+static void init_threads(void) {
```

Comment:
If you still need to mark something "constructor" can you elaborate on what went wrong with LTO?

I'm sure we can find an attribute that should/shouldn't be set to fix it.

However, I think if you init the serial struct as a global you won't need to make any changes to this file after all.